ocs_ci/ocs/scale_noobaa_lib.py

```diff
@@ -42,50 +45,50 @@ def construct_obc_creation_yaml_bulk_for_kube_job(no_of_obc, sc_name, namespace)
 
 
 def check_all_obc_reached_bound_state_in_kube_job(
-    kube_job_obj, namespace, no_of_obc, timeout=120, no_wait_time=20
+    kube_job_obj, namespace, no_of_obc, timeout=60, no_wait_time=10
 ):
     """
     Function to check either bulk created OBCs reached Bound state using kube_job
-
     Args:
         kube_job_obj (obj): Kube Job Object
         namespace (str): Namespace of OBC's created
         no_of_obc (int): Bulk OBC count
         timeout (second): a timeout for all the obc in kube job to reach bound state
         no_wait_time (int): number of wait time to ensure all OCBs to reach bound state
-
     Returns:
         obc_bound_list (list): List of all OBCs which is in Bound state.
-
     Raises:
         AssertionError: If not all OBC reached to Bound state
-
     """
     # Check all OBCs to reach Bound state
     obc_bound_list, obc_not_bound_list = ([] for i in range(2))
-    while_iteration_count = 0
+    while_iteration_count_1 = 0
+    while_iteration_count_2 = 0
     while True:
-        # Get kube_job obj and fetch either all OBC's are in Bound state
+        # Get kube_job obj and check if all OBC's are in Bound state
         # If not bound adding those OBCs to obc_not_bound_list
-        job_get_output = kube_job_obj.get(namespace=namespace)
+        job_get_output = kube_job_obj.get(namespace=namespace).get("items")
         if job_get_output is not None and len(job_get_output) == no_of_obc:
             for i in range(no_of_obc):
-                status = job_get_output["items"][i]["status"]["phase"]
-                log.info(
-                    f"obc {job_get_output['items'][i]['metadata']['name']} status {status}"
-                )
-                if status != constants.STATUS_BOUND:
-                    obc_not_bound_list.append(
-                        job_get_output["items"][i]["metadata"]["name"]
-                    )
+                status = job_get_output[i]["status"]["phase"]
+                log.info(f"obc {job_get_output[i]['metadata']['name']} status {status}")
+                if not status or status != constants.STATUS_BOUND:
+                    obc_not_bound_list.append(job_get_output[i]["metadata"]["name"])
+                    # Wait 20 secs to ensure the next obc on the list has status field populated
+                    time.sleep(20)
```

Comment:
Please change to sampling the existence of the next obc on the list to be sure it is populated

Reply:
will address this in next PR.